engine/packages/guard-core/src/proxy_service.rs

```diff
@@ -1895,18 +1895,33 @@ impl ProxyService {
 								Err(err) => {
 									tracing::debug!(?err, "websocket handler error");
 
-									// Denotes that the connection did not fail, but needs to be retried to
-									// resole a new target
-									let ws_retry = is_ws_retry(&err);
+									// Denotes that the connection did not fail, but the downstream has closed
+									let ws_hibernate = is_ws_hibernate(&err);
 
-									if ws_retry {
+									if ws_hibernate {
 										attempts = 0;
 									} else {
 										attempts += 1;
 									}
 
-									if attempts > max_attempts
-										|| (!is_retryable_ws_error(&err) && !ws_retry)
+									if ws_hibernate {
+										// After this function returns false:
+										// - the route will be resolved again
+										// - the websocket will connect to the new downstream target
+										// - the gateway will continue reading messages from the client ws
+										//   (starting with the message that caused the hibernation to end)
+										let close = handler
+											.handle_websocket_hibernation(ws_handle.clone())
+											.await?;
+
+										if close {
+											tracing::debug!(
+												"hibernating websocket handler complete"
```

Comment:
we need to wake & send close message to the actor on client close, even if the actor is sleeping.